proto/frontend/src/action_prof_mgr.cpp

```diff
@@ -57,29 +57,18 @@ using OneShotMember = ActionProfAccessOneshot::OneShotMember;
 
 namespace {
 
-// temporary until we have port translation support
-
-int bytestring_to_integer(const std::string &str) {
-  int v = 0;
-  if (str.size() > 4) return -1;
-  for (auto c : str) {
-    v = v << 8;
-    v += static_cast<int>(static_cast<unsigned char>(c));
-  }
-  return v;
-}
-
 pi_port_t watch_port_p4rt_to_pi(int watch) {
   return (watch == p4v1::SDN_PORT_UNKNOWN) ?
       WatchPortEnforcer::INVALID_WATCH : static_cast<pi_port_t>(watch);
 }
 
 pi_port_t watch_port_p4rt_to_pi(const std::string &watch) {
   if (watch == "") return WatchPortEnforcer::INVALID_WATCH;
-  auto v = bytestring_to_integer(watch);
-  if (v < 0 || v == p4v1::SDN_PORT_UNKNOWN)
+  pi_port_t v;
+  Status status = common::bytestring_to_pi_port(watch, &v);
+  if (!IS_OK(status) || v < 0 || v == p4v1::SDN_PORT_UNKNOWN)
```

Comment:
The `v < 0` check doesn't seem meaningful anymore now that `bytestring_to_pi_port` returns a `Status`?

Reply:
`bytestring_to_pi_port` allows arbitrary byte strings of size <= 4, essentially any `uint32_t`. That decision was loosely based on this part of the spec

![image](https://github.com/p4lang/PI/assets/6642034/1d6d7ae8-2842-4fef-b44f-4f7816981907)

which includes SDN port numbers where the highest bit is set.

Perhaps we want to be more restrictive and pull all these checks into `bytestring_to_pi_port`? I just wasn't sure what the right restrictions were. Happy to take suggestions here.

Reply:
It may make sense to have these checks in `bytestring_to_pi_port` then, because `GetReplicaPort` also calls `bytestring_to_pi_port` and does not validate the value right now (only the returned status).

Reply:
Done.